packages/reactivue/src/component.ts

```diff
@@ -4,13 +4,34 @@ import { isDev } from './env'
 import { invokeLifeCycle } from './lifecycle'
 import { InternalInstanceState, LifecycleHooks } from './types'
 
-let _id = 0
-const _vueState: Record<number, InternalInstanceState> = {}
+const isClient = typeof window !== 'undefined'
+
+type ReactiveEffects = Record<number, InternalInstanceState>
+
+/**
+ * When `reactivue` dependency gets updated during developmment
+ * your build tool re-executes it and `_vueState` becomes its
+ * initial state. Storing our reactive efects in `window.__reactivue_state`
+ * and filling our `_vueState` with it.
+ */
+declare global {
+  interface Window { __reactivue_state: ReactiveEffects }
+}
+
+const _vueState: ReactiveEffects = (isDev && isClient && window.__reactivue_state) || {}
 
 export let currentInstance: InternalInstanceState | null = null
 export let currentInstanceId: number | null = null
 
-export const getNewInstanceId = () => _id++
+export const getNewInstanceId = () => {
+  // When React is in Strict mode, it runs state functions twice
+  // Remove unmounted instances before creating new one
+  if (isDev)
+    Object.keys(_vueState).forEach(id => !_vueState[+id].isActive && unmount(+id))
```

Comment:
```suggestion
    Object.keys(_vueState).forEach(id => {
      if (!_vueState[id].isActive)
        unmount(id)
    })
```

It's better to use `&&` for value operation and `if` for effects.